hw/ip/keymgr/doc/_index.md

```diff
@@ -158,6 +158,32 @@ This means the working states, even though functionally 256b, are maintained as
 For advance-state and `generate-output` commands, the KMAC emitted output are also in 2-shares.
 Software is responsible for determining if the key should be preserved in shares or combined, depending on the use case.
 
+## Errors, Interrupts and Alerts
+An error code register is mainteined {{< regref ERR_CODE >}} to check issues that might rise while using the key manager.
+There are two categories of errors
+*  Hardware fault errors - These errors indicate something fundamental has gone wrong and are errors that could not have been caused by software.
+   *  Invalid command - A non-one-hot command was issued from the key manager controller to the KMAC data interface. This is not possible by software and indicates a hardware fault.
+   *  Invalid output - The data return from KMAC is all 0's or all 1's.  This is not possible given the set of KMAC data interface inputs.
+
+*  Software operation errors - These errors could have been caused by user errors and is a sign that software should examine its usage of key manager.
+   *  Invalid operation - An invalid operation (for example `generate` while in Idle) was invoked.
+   *  Invalid input - Invalid software input was supplied (for example a greater key version than allowed in {{< regref MAX_OWNER_KEY_VER >}}.
+
```

Comment:
could you also document when error occurs, what the outcome is. If I understand correctly, below will happen.
Invalid operation: only generate alert, interrupt
Invalid input: besides alert and interrupt, corrupt kmac output data and advance state to StDisabled
Invalid command: besides alert and interrupt, corrupt kmac output data and advance state to StDisabled
Invalid output: besides alert and interrupt, KMAC key is disabled

Reply:
@weicaiyang i updated the document to make this area clearer.
I also noticed a couple of inconsistencies in the design as well.